src/cacheout/stats.py

```diff
@@ -0,0 +1,166 @@
+from threading import RLock
+
+
+class Stats:
+    """An object to represent a snapshot of statistics."""
+
+    def __init__(self, hits: int, misses: int, evictions: int, total: int) -> None:
+        self._hits = hits
+        self._misses = misses
+        self._evictions = evictions
+        self._total = total
```

Comment:
Match naming with `def total_entries`:

```suggestion
    def __init__(self, hits: int, misses: int, evictions: int, total_entries: int) -> None:
        self._hits = hits
        self._misses = misses
        self._evictions = evictions
        self._total_entries = total_entries
```